### PR TITLE
Add signature to pyfunction

### PR DESCRIPTION
**Pull request by sanders41**
I made a slight tweak here to add a signature to the pyfunction based on something I learned in David's latest example https://github.com/davidhewitt/dict-not-none. Doing this let me simplify the Rust code and additionally use named parameters.

### CODE REVIEW

**Comment by prrao87** [comment]
I see the chain haha. I presume you saw my star on this repo in your feed? I saw Samuel Colvin start this yesterday himself in mine, and found it interesting. 🤣

**Comment by prrao87** [comment]
I didn't know this syntax worked all the way back to Python 3.8! Is that correct?

**Comment by sanders41** [comment]
> I see the chain haha. I presume you saw my star on this repo in your feed? I saw Samuel Colvin start this yesterday himself in mine, and found it interesting. 🤣

I saw the star from Samuel also, then saw you stared it also :smile:

File: src/pyo3_mock_data/pyo3_mock_data/_pyo3_mock_data.pyi [hunk 1]
```diff
@@ -3,5 +3,5 @@ from __future__ import annotations
 from pathlib import Path
 
 def generate_fake_persons(
```
Comment (prrao87) [inline src/pyo3_mock_data/pyo3_mock_data/_pyo3_mock_data.pyi:5]: Can we call this "mock" persons instead? We should remove all mentions of the word "fake" and be consistent with the parent direct name.

**Review by prrao87 — CHANGES_REQUESTED** [review]

**Comment by sanders41** [comment]
> I didn't know this syntax worked all the way back to Python 3.8! Is that correct?

You mean named only? They were added in 3.8.

File: src/pyo3_mock_data/pyo3_mock_data/_pyo3_mock_data.pyi (hunk 1 above)
Comment (sanders41) [reply src/pyo3_mock_data/pyo3_mock_data/_pyo3_mock_data.pyi:5]: Yes, I thought I caught all of these, but apparently I missed some. Will do a global search to make sure I didn't miss more.

Comment (sanders41) [reply src/pyo3_mock_data/pyo3_mock_data/_pyo3_mock_data.pyi]: wow, looks like the one I missed just the function name...the most important one.

**Pull request merged**